enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1334 @@
+    name: <desired rendered MachineConfig>
+    validation: None
+```
+
+The manual strategy requests no automated initiation of updates. New and rebooting
+nodes will only receive the desired configuration. From a metrics perspective, this strategy
+is always paused state.
+
+#### Metrics
+
+`change_management_change_pending`
+Labels:
+- kind=ClusterVersion|MachineConfigPool|HostedCluster|NodePool
+- object=<object-name>
+- system=<control-plane|worker-nodes>
+
+Value: 
+- `0`: no material changes are pending.
+- `1`: changes are pending but being initiated.
+- `2`: changes are pending and blocked based on this resource's change management policy.
+
+`change_management_next_change_eta`
+Labels:
+- kind=ClusterVersion|MachineConfigPool|HostedCluster|NodePool
+- object=<object-name>
+- system=<control-plane|worker-nodes>
+
+Value: 
+- `-2`: Error determining the time at which changes can be initiated (e.g. cannot check with ClusterVersion / change management hierarchy). 
+- `-1`: Material changes are paused indefinitely.  
+- `0`: Material changes can be initiated now (e.g. change management is disabled or inside machine schedule window). 
+  Alternatively, time is not relevant to the strategy (e.g. restrictive).
+- `> 0`: The number seconds remaining until changes can be initiated.
+
+`change_management_permissive_remaining`
+Labels:
+- kind=ClusterVersion|MachineConfigPool|HostedCluster|NodePool
+- object=<object-name>
+- system=<control-plane|worker-nodes>
+
+Value: 
+- `-2`: Error determining the time at which current permissive window will close.
+- `-1`: Material changes are permitted indefinitely (e.g. `strategy: Permissive`). 
+- `0`: Material changes are not presently permitted (i.e. the cluster is outside of a permissive window). 
+  Alternatively, time is not relevant to the strategy (e.g. restrictive strategy).
+- `> 0`: The number seconds remaining in the current permissive change window. 
+
+`change_management_last_change`
+Labels:
+- kind=ClusterVersion|MachineConfigPool|HostedCluster|NodePool
+- object=<object-name>
+- system=<control-plane|worker-nodes>
+
+Value: 
+- `-1`: Datetime unknown. 
+- `0`: Material changes are currently permitted.
+- `> 0`: The number of seconds which have elapsed since the material changes were last permitted or initiated.
+
+`change_management_strategy_enabled`
+Labels:
+- kind=ClusterVersion|MachineConfigPool|HostedCluster|NodePool
+- object=<object-name>
+- system=<control-plane|worker-nodes>
+- strategy=MaintenanceSchedule|Permissive|Restrictive
+
+Value: 
+- `0`: Change management for this resource is not subject to this enabled strategy.
+- `1`: Change management for this resource is subject to this enabled strategy.
+
+#### Change Management Bypass Annotation
+In some situations, it may be necessary for a MachineConfig to be applied regardless of the active change
+management policy for a MachineConfigPool. In such cases, `machineconfiguration.openshift.io/bypass-change-management`

```

Comment:
Does the API not already have ways to bypass the change management? Why not leverage those? Or are we now looking granularly at the MachineConfig level? I think we'll have to square that with the MCO team as I don't think that's a feature they have today

Reply:
There are several ways to bypass change management. This mechanism was a way to make "control-plane as a single material change" less of a snowflake (i.e. it happens as a single material change because of a standard annotation vs just being hard coded).
It also gives SRE some guarantees for critical machine config updates without having to guess at a "disabledUntil" date.

Reply:
> Rolling out critical machine config changes for worker nodes also is made easier with this annotation. 
Instead of, for example, trying to predict a `disabledUntil` date, an SRE/operations team can use this 
annotation to specify their goal with precision. Compare this with `disabledUntil`, which 
(a) an operations team would generally overestimate in order to ensure that updates complete and 
(b) may cause subsequent, non-critical, machine configuration changes to cause further workload disruption
(node reboots) that are unwarranted.